lib/OpenQA/Jobs/Constants.pm

```diff
@@ -51,8 +51,9 @@ use constant STATES => (SCHEDULED, ASSIGNED, SETUP, RUNNING, UPLOADING, DONE, CA
 # "meta" states
 use constant PENDING_STATES       => (SCHEDULED, ASSIGNED, SETUP,   RUNNING, UPLOADING);
 use constant EXECUTION_STATES     => (ASSIGNED,  SETUP,    RUNNING, UPLOADING);
-use constant PRE_EXECUTION_STATES => (SCHEDULED);        # Assigned belongs to pre execution, but makes no sense for now
-use constant FINAL_STATES         => (DONE, CANCELLED);
+use constant PRE_EXECUTION_STATES => (SCHEDULED);
+use constant PRISTINE_STATES      => (SCHEDULED, ASSIGNED);    # no worker reported any updates/results so far
```

Comment:
hm, did you think about just relying on `PRE_EXECUTION_STATES`? I see this hardly used in our code so maybe you can just use that instead of defining new states definition.

Reply:
But we use `EXECUTION_STATES` a lot and there `ASSIGNED` is supposed to be included. It makes no sense to have `ASSIGNED` in both, `PRE_EXECUTION_STATES` and `EXECUTION_STATES` so I would rather keep `PRISTINE_STATES`.

Reply:
ok